pkg/provider/cloud/openstack/provider.go

```diff
@@ -164,6 +164,21 @@ func validateExistingSubnetOverlap(networkID string, netClient *gophercloud.Serv
 	})
 }
 
+func ensureFinalizers(cluster *kubermaticv1.Cluster, finalizers []string, update provider.ClusterUpdater) (*kubermaticv1.Cluster, error) {
+	var err error
+
+	if len(finalizers) > 0 {
+		cluster, err = update(cluster.Name, func(cluster *kubermaticv1.Cluster) {
+			kubernetes.AddFinalizer(cluster, finalizers...)
+		}, provider.UpdaterOptionOptimisticLock)
+		if err != nil {
+			return nil, fmt.Errorf("failed to add finalizers: %w", err)
+		}
```

Comment:
This can simply be `return update(.....)`. Otherwise, in case of an error, the full error message will be "failed to ensure finalizers: failed to add finalizers: ...", which is kinda redundant.

Reply:
it cannot :P becuase then in the `InitializeCloudProvider` function i would have condition to `if len(finalizers)...`  which will cause `pkg/provider/cloud/openstack/provider.go:169:1: cyclomatic complexity 31 of func `(*Provider).InitializeCloudProvider` is high (> 30) (gocyclo)`

with the error message I agree

Reply:
@xrstf what do you think to add gocyclo:ignore directive and do a proper refactoring later? But as always who will have time to do such a thing later....

Reply:
I fixed the message by not wrapping the error in the ensureFinalizers func